From: Matteo Percivaldi <m.percivaldi@fastassembler.com>
Date: Fri, 13 Jan 2023 15:19:37 -0500
Subject: gtkcssanimatedstyle: Fix return of new_advance()

This was causing animation and transition to stop randomly and reset
their state to initial state.

This issue has existed since commit
7b68bdb8316fc1bb96c4a5ad16c1885506131d22.

Bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/4426
Origin: upstream, 3.24.37, commit:6a2fb84446
---
 gtk/gtkcssanimatedstyle.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkcssanimatedstyle.c b/gtk/gtkcssanimatedstyle.c
index 3fa005f..3c65328 100644
--- a/gtk/gtkcssanimatedstyle.c
+++ b/gtk/gtkcssanimatedstyle.c
@@ -466,9 +466,12 @@ gtk_css_animated_style_new_advance (GtkCssAnimatedStyle *source,
   gtk_internal_return_val_if_fail (GTK_IS_CSS_ANIMATED_STYLE (source), NULL);
   gtk_internal_return_val_if_fail (GTK_IS_CSS_STYLE (base), NULL);
   
-  if (timestamp == 0 || timestamp == source->current_time)
+  if (timestamp == 0)
     return g_object_ref (source->style);
 
+  if (timestamp == source->current_time)
+    return g_object_ref (GTK_CSS_STYLE (source));
+
   gtk_internal_return_val_if_fail (timestamp > source->current_time, NULL);
 
   animations = NULL;
